src/ExpressionParser.php

```diff
@@ -432,6 +432,52 @@ public function parseMappingExpression()
         return $node;
     }
 
+    /*
+     * Unlike its sibling methods above, method requires a TokenStream parameter.
+     * 
+     * The reason is that calling `$this->parser->getStream()` (as in `parseMappingExpression()`)
+     * only works _while_ `parser->parse()` is running, because that method sets and unsets the $stream property.
+     * 
+     * The upside is that this method is fully pure, since it doesn't depend on the state of $this.
+     */
+    public function parseNameToStringMapping(TokenStream $stream, bool $allowOptionalKeys): ArrayExpression
```

Comment:
This `$allowOptionalKeys` parameter should be removed, as it is always used as `true`